crypto/src/asset.rs

```diff
@@ -42,11 +46,66 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_displaydenom_format_value() {
+        // with exponent 6, 1782000 formats to 1.782
+        let penumbra_display_denom = REGISTRY.parse_display("penumbra").unwrap();
+        assert_eq!(penumbra_display_denom.format_value(1782000), "1.782");
+
+        // with exponent 3, 1782000 formats to 1782
+        let mpenumbra_display_denom = REGISTRY.parse_display("mpenumbra").unwrap();
+        assert_eq!(mpenumbra_display_denom.format_value(1782000), "1782");
+
+        // with exponent 0, 1782000 formats to 1782000
+        let upenumbra_display_denom = REGISTRY.parse_display("upenumbra").unwrap();
+        assert_eq!(upenumbra_display_denom.format_value(1782000), "1782000");
+    }
+
+    #[test]
+    fn test_displaydenom_parse_value() {
+        let penumbra_display_denom = REGISTRY.parse_display("penumbra").unwrap();
+        assert!(penumbra_display_denom.parse_value("1.2.3").is_err());
+
+        assert_eq!(
+            penumbra_display_denom.parse_value("1.782").unwrap(),
+            1782000
+        );
+
+        let mpenumbra_display_denom = REGISTRY.parse_display("mpenumbra").unwrap();
+        assert_eq!(
+            mpenumbra_display_denom.parse_value("1782").unwrap(),
+            1782000
+        );
```

Comment:
Could be nice to check that
```
mpenumbra_display_denom.parse_value("1782.0001").is_err()
```
just to exercise the too-much-precision codepath.